### ISSUE TITLE
Certain graphical elements update at very low rates unless application window move is in progress

### ISSUE THREAD

**Issue by Von-Cheam** [Vladar4/nimgame2#36]
This is hard to explain, so I've made a couple of short screencasts [here](https://sendvid.com/7brlru2i) and [here](https://sendvid.com/eyzaztqa) to demonstrate what I mean. Essentially, certain elements (entities?) update much, much slower than expected (based on the framerate, etc). Even stranger, moving the application window causes the update rate of these elements to increase significantly.

Happy to provide details of my system, or do any testing you might require, if it'll help!

**Comment by Vladar4** [comment]
And I assume the source code wasn't modified in any way?
This is odd behavior indeed, never encountered such occurrence before. The data on your system might help though it would be pretty had to debug if I couldn't recreate this myself.

**Comment by Von-Cheam** [comment]
Yeah, the source wasn't touched at all - I just installed via Nimble (tried both devel and head branches.)

I'm uploading some system info - if there's any other system data that might help, I'd be happy to let you know.

[system_info.txt](https://github.com/Vladar4/nimgame2/files/3876496/system_info.txt)

**Comment by Von-Cheam** [comment]
Well, er, this is embarrassing! Have been trying to progressively change/disable things, to try and figure out what's going on; changing from the Nvidia video card to onboard Intel graphics fixed the problem. Why that should be the case, I have no idea, but I think we could probably close this now, unless you want me to investigate further or anything?

**Comment by Vladar4** [comment]
Thanks for the info!

Honestly, right now I have no idea of what could have cause this bizarre behavior, especially considering that I've been developing the engine on nvidia-powered machine for a long time, before switching to AMD last year. Do you think it could be a driver-related issue?

In any case, please let me know if you figure it out in the future.
One of the points of testing might be the update call in main loop: https://github.com/Vladar4/nimgame2/blob/master/nimgame2/nimgame.nim#L433 , e.g. is it slows down as well, or the issue is purely render-related.

**Comment by Von-Cheam** [comment]
I found it pretty strange, too! Yeah; if I had to guess, my first guess would be something in SDL2 not playing nicely with Nvidia Linux drivers. (My drivers are pretty recent, I think, but, er, that doesn't exactly mean much, with Nvidia..)

Thanks for the tip - don't think the main loop is slowing down, as text-rendering doesn't seem to be affected, just textures, graphics primitives, etc.

Will let you know if I figure anything out - and in the mean time I'll stick to Intel graphics!

**Comment by Vladar4** [comment]
Thanks!
